# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1532617818 14400
#      Thu Jul 26 11:10:18 2018 -0400
# Node ID a6af9578c18a8e0fa996b8fe5bf46c63e3a2e684
# Parent  20aa564402c441b85748d0464dd814d8d1781365
# Parent  1ba60c0455f4cc8ae685db76982d1241e9a9f537
maint: Merge stable to default.

diff --git a/liboctave/system/oct-uname.cc b/liboctave/system/oct-uname.cc
--- a/liboctave/system/oct-uname.cc
+++ b/liboctave/system/oct-uname.cc
@@ -37,17 +37,17 @@ namespace octave
     void
     uname::init (void)
     {
       char *sysname, *nodename, *release, *version, *machine;
 
       err = octave_uname_wrapper (&sysname, &nodename, &release,
                                   &version, &machine);
 
-      if (err)
+      if (err < 0)
         msg = std::strerror (errno);
       else
         {
           m_sysname = sysname;
           m_nodename = nodename;
           m_release = release;
           m_version = version;
           m_machine = machine;
diff --git a/liboctave/wrappers/uname-wrapper.c b/liboctave/wrappers/uname-wrapper.c
--- a/liboctave/wrappers/uname-wrapper.c
+++ b/liboctave/wrappers/uname-wrapper.c
@@ -37,27 +37,27 @@ along with Octave; see the file COPYING.
 int
 octave_uname_wrapper (char **sysname, char **nodename,
                       char **release, char **version, char **machine)
 {
   static struct utsname unm;
 
   int err = uname (&unm);
 
-  if (err == 0)
+  if (err < 0)
+    {
+      *sysname = 0;
+      *nodename = 0;
+      *release = 0;
+      *version = 0;
+      *machine = 0;
+    }
+  else
     {
       *sysname = unm.sysname;
       *nodename = unm.nodename;
       *release = unm.release;
       *version = unm.version;
       *machine = unm.machine;
     }
-  else
-    {
-      *sysname = 0;
-      *nodename = 0;
-      *release = 0;
-      *version = 0;
-      *machine = 0;
-    }
 
   return err;
 }
